plugins/blip/basics/chartbasicsfactory.js

```diff
@@ -89,7 +92,47 @@ var BasicsChart = React.createClass({
       basicsData = _.assign({}, basicsData, _.cloneDeep(basicsState));
       var dataMunger = dataMungerMkr(this.props.bgClasses);
       dataMunger.reduceByDay(basicsData);
-      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData);
+
+      // get site changes from three event types
+      basicsData.data.cannulaPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_CANNULA);
+      basicsData.data.tubingPrime.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_TUBING);
+      basicsData.data.reservoirChange.infusionSiteHistory = dataMunger.infusionSiteHistory(basicsData, constants.SITE_CHANGE_RESERVOIR);
+
+      var siteChangeEvents = [];
+
+      if (basicsData.data.tubingPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.tubingPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.cannulaPrime.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.cannulaPrime.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+      if (basicsData.data.reservoirChange.infusionSiteHistory.hasChangeHistory) {
+        siteChangeEvents.push(_.last(_.findLast(basicsData.data.reservoirChange.infusionSiteHistory, {type: constants.SITE_CHANGE}).data));
+      }
+
+      if (siteChangeEvents.length) {
+        // get latest pump upload:
+        var latestPump = _.last(_.sortBy(siteChangeEvents, 'time')).source;
```

Comment:
I believe we do have code to get the most recent pump settings (via the most recent upload data, IIRC) for the device settings page. Perhaps reference that?

What you have here does not look like "latest pump" - it looks like most recent site change event, and that's not the same thing (imagine a user who switches back and forth between a MedT with no site change data and another pump that does have site change data - this could result in a mismatch between "most recent pump" and "most recent site change data"), and having just reviewed the Trello card it appears the requirement we settled on was to use the most recent pump upload as the determinant of what's initially shown.

Also could you perhaps move most/all of this new code into the `infusionSiteHistory` function in the datamunger? It all feels like the same preprocessing task to me, or is there something that I'm missing? That way you should be able to write a lot more tests to isolate the various device situations without dealing with testing on a React component.